### PR TITLE
Remove tags and branch heads sync.

### PR DESCRIPTION
**Pull request by goodov**
We don't need to fetch all tags and heads, but we had to do this, because it was impossible to sync a specific Chromium tag without those.

`gclient sync` is now fixed and can sync any tag without explicitly fetching everything.
https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/dbbe224787188ca748eef54e44d6651c1c90b90b

Related https://github.com/brave/brave-browser/issues/35512

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/lib/syncUtils.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ function syncChromium(syncWithForce, sync_chromium, delete_unused_deps) {
   const requiredChromiumRef = config.getProjectRef('chrome')
   let args = [
     'sync', '--nohooks', '--revision', 'src@' + requiredChromiumRef, '--reset',
-    '--with_tags', '--with_branch_heads', '--upstream'
```
Comment (goodov) [inline build/commands/lib/syncUtils.js:133]: @cdesouza-chromium I was running all my working directories in this setup for a week now and haven't experienced any issues. I also did a clean Brave checkout with a an empty git cache, things were good.

Can you please try this change manually in your `canary` branch flow for a few days? If Chromium bumps won't trigger any `gclient sync` issues, I believe we can merge this safely.

Comment (cdesouza-chromium) [reply build/commands/lib/syncUtils.js:133]: Will pull this into Canary. No worries.

Comment (atuchin-m) [reply build/commands/lib/syncUtils.js:133]: If I'm not mistaken that changes results to inability to sync some chromium revisions.
The issues was that some third_party HEADs were set to those branch_heads and `gclient sync` doesn't  any code to fetch them.
I hope the situation is changed now.

The instruction still referrers the switches for working with release branches:
https://www.chromium.org/developers/how-tos/get-the-code/working-with-release-branches/

Comment (atuchin-m) [reply build/commands/lib/syncUtils.js:133]: My old attempt to remove them: https://github.com/brave/brave-core/pull/11526#discussion_r787843987

Comment (goodov) [reply build/commands/lib/syncUtils.js:133]: > If I'm not mistaken that changes results to inability to sync some chromium revisions. The issues was that some third_party HEADs were set to those branch_heads and `gclient sync` doesn't any code to fetch them. I hope the situation is changed now.

yep, this exact issue should be fixed now:
https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/dbbe224787188ca748eef54e44d6651c1c90b90b

**Review by atuchin-m — APPROVED** [review]

File: build/commands/lib/syncUtils.js (hunk 1 above)
Comment (goodov) [reply build/commands/lib/syncUtils.js:133]: > Will pull this into Canary. No worries.

@cdesouza-chromium I assume there were no issues?

Comment (cdesouza-chromium) [reply build/commands/lib/syncUtils.js:133]: I have seen any issues so far, yeah.

**Pull request merged**